examples/README.md

```diff
@@ -25,6 +25,9 @@ before trying out the examples.
 
 - [Json serialization](src/main/java/io/grpc/examples/advanced)
 
+- [Authentication](src/main/java/io/grpc/examples/authentication)
```

Comment:
Just a heads-up: if you would like to add detailed document, you can place it at `examples/AUTHENTICATION_EXAMPLE.md` and link it like
```
[Authentication](AUTHENTICATION_EXAMPLE.md)
``` 

And point to the code location `src/main/java/io/grpc/examples/authentication` in  `AUTHENTICATION_EXAMPLE.md`.

Reply:
Thanks. Will do

Reply:
@dapengzhang0  done. pls check